segger-jlink: 810 -> 824, add Darwin support

siriobalmelli opened the pull request:
- Update segger-jlink 810 -> 824
- Add Darwin support

cc @NixOS/darwin-maintainers 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Emin017 reviewed (approved):
LGTM.

usertam reviewed (approved):
https://github.com/usertam/nixpkgs-review-actions/actions/runs/14355007262

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review --extra-nixpkgs-config '{ allowUnfree = true; segger-jlink.acceptLicense = true; permittedInsecurePackages = [ "segger-jlink-qt4-824" ]; }'`

### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>nrf-command-line-tools</li>
    <li>nrfconnect</li>
    <li>nrfconnect-bluetooth-low-energy</li>
    <li>nrfutil</li>
    <li>segger-jlink</li>
    <li>segger-jlink-headless</li>
  </ul>
</details>

### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>nrf-command-line-tools</li>
    <li>nrfconnect</li>
    <li>nrfconnect-bluetooth-low-energy</li>
    <li>segger-jlink</li>
    <li>segger-jlink-headless</li>
  </ul>
</details>

### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nrfutil</li>
    <li>segger-jlink (segger-jlink-headless)</li>
  </ul>
</details>

### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nrfutil</li>
    <li>segger-jlink (segger-jlink-headless)</li>
  </ul>
</details>

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211`

drupol commented on pkgs/by-name/se/segger-jlink/package.nix:
```diff
@@ -55,118 +60,169 @@ let
+    ];
 
-      # Link executables into /bin/
+    unpackPhase = ''
```
If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

siriobalmelli replied:
ok, respinning

drupol commented on pkgs/by-name/se/segger-jlink/package.nix:
```diff
@@ -55,118 +60,169 @@ let
+      xar -xf $src
+    '';
+
+    installPhase = ''
```
Same here.

siriobalmelli replied:
ok, respinning

drupol commented on pkgs/by-name/se/segger-jlink/package.nix:
```diff
@@ -55,118 +60,169 @@ let
+
+    passthru.updateScript = ./update.py;
+
+    meta = with lib; {
```
```suggestion
    meta = {
```

siriobalmelli replied:
ok, respinning

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211`
Commit: `1fbf7675ab3bd0a23fb95b38beb4e6ab316bd371`

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211 --package segger-jlink`
Commit: `1fbf7675ab3bd0a23fb95b38beb4e6ab316bd371`

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>segger-jlink</li>
  </ul>
</details>

sarahec commented:
@siraben that's because the package needs an attribute to be set (`segger-jlink.acceptLicense = true`) before it will build on any platform.

siraben commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211`
Commit: `1fbf7675ab3bd0a23fb95b38beb4e6ab316bd371`

siriobalmelli committed:
segger-jlink: refactor source.nix and update.py

Adds parameters necessary to add Darwin sources in a later commit.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
segger-jlink: 810 -> 824

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
segger-jlink: add Darwin support

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
segger-jlink: refactor build attributes in preparation for adding Darwin

Isolate linux-specific build attributes.
Provide a mechanism for specifying Darwin attributes.

No functional change.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
segger-jlink: remove 'with lib;' antipattern from meta

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211 --extra-nixpkgs-config '{ allowUnfree = true; segger-jlink.acceptLicense = true; permittedInsecurePackages = [ "segger-jlink-qt4-824" ]; }'`
Commit: `b27f4ff50d50feb4e50c325cec093c9054e95c31`

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211 --extra-nixpkgs-config '{ allowUnfree = true; segger-jlink.acceptLicense = true; permittedInsecurePackages = [ "segger-jlink-qt4-824" ]; }'`
Commit: `b27f4ff50d50feb4e50c325cec093c9054e95c31`

siriobalmelli commented:
Would be great to get this merged 🙏

siriobalmelli commented:
Bump

sarahec commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211`
Commit: `b27f4ff50d50feb4e50c325cec093c9054e95c31`

sarahec commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397211 --extra-nixpkgs-config '{ allowUnfree = true; segger-jlink.acceptLicense = true; permittedInsecurePackages = [ "segger-jlink-qt4-824" ]; }'`
Commit: `b27f4ff50d50feb4e50c325cec093c9054e95c31`

h7x4 commented:
I wonder why ci didn't autorequest maintainer reviews

h7x4 reviewed (approved):
I'm not too familiar with the darwin stuff, but the diff looks fine and the package seems to work well on x86_64-linux.

sarahec reviewed (approved):
Builds and runs on aarch64-Darwin. LGTM.
